tests/org.eclipse.reddeer.eclipse.test/src/org/eclipse/reddeer/eclipse/test/wst/server/ui/RuntimePreferencePageTest.java

```diff
@@ -111,4 +114,39 @@ public void editRuntime(){
 
 		assertFalse(new PushButton(RuntimePreferencePage.EDIT_BUTTON_NAME).isEnabled());
 	}
+	
+	@Test
+	public void getServerRuntime() {
+		NewRuntimeWizardDialog wizardDialog = preferencePage.addRuntime(); 
+		NewRuntimeWizardPage wizardPage = new NewRuntimeWizardPage(wizardDialog);
+		wizardPage.selectType(SERVER_PATH, SERVER_NAME);
+		wizardDialog.finish();
+		
+		wizardDialog = preferencePage.addRuntime();
+		wizardPage = new NewRuntimeWizardPage(wizardDialog);
+		wizardPage.selectType(SERVER_PATH, SERVER_NAME_2);
+		wizardDialog.finish();
+		
+		wizardDialog = preferencePage.addRuntime();
+		wizardPage = new NewRuntimeWizardPage(wizardDialog);
+		wizardPage.selectType(SERVER_PATH, SERVER_NAME_2);
+		wizardDialog.finish();
+		
+		List<Runtime> runtimes = preferencePage.getServerRuntimes();
+		assertThat(runtimes.size(), is(3));
+		
+		Runtime runtime1 = preferencePage.getServerRuntime(SERVER_NAME, SERVER_NAME);
+		assertThat(runtime1, is(new Runtime(SERVER_NAME, SERVER_NAME)));
+		
+		Runtime runtime2 = preferencePage.getServerRuntime(new Runtime(SERVER_NAME_2, SERVER_NAME_2));
+		assertThat(runtime2, is(new Runtime(SERVER_NAME_2, SERVER_NAME_2)));
+		
+		Runtime runtime3 = null;
+		try {
+			runtime3 = preferencePage.getServerRuntime(SERVER_NAME_2);
+		} catch (IllegalStateException exc) {
```

Comment:
I guess EclipseLayerEx should be in the catch block

Reply:
Oh dear, you are right... I have not had a coffee yet... Thanks